### PR TITLE
Move migration guide to Docc

### PR DESCRIPTION
**Pull request by aboedo**
Moves migration guide to Docc. This helps fix links easily, since Xcode checks them and issues build warnings if anything goes wrong.

### CODE REVIEW

File: Documentation.docc/V4_API_Migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+### ObjC type changes
+
+`@import Purchases` is now `@import RevenueCat`
+
+| v3 | v4 |
+| ------------ | ------------------------------------- | 
+| RCPurchaserInfo | RCCustomerInfo |
+| RCTransaction | RCStoreTransaction |
+| RCPackage.product | RCPackage.storeProduct |
+| (RCPurchasesErrorCode).RCOperationAlreadyInProgressError | RCOperationAlreadyInProgressForProductError |
```
Comment (aboedo) [inline Documentation.docc/V4_API_Migration_guide.md:65]: This is the only change in this file - updating the table format from HTML format to markdown, which Docc supports

Comment (aboedo) [reply Documentation.docc/V4_API_Migration_guide.md:65]: it should look the same, though

File: Documentation.docc/V4_API_Migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+| purchaserInfo(_ completion:) | ``Purchases/getCustomerInfo(completion:)`` |
+| offerings(_ completion:) | ``Purchases/getOfferings(completion:)`` |
+| products(_ productIdentifiers:, _ completion:) | ``Purchases/getProducts(_:completion:)`` |
+| purchaseProduct(_ product:, _ completion:) | ``Purchases/purchase(product:completion:)`` |
+| purchasePackage(_ package:, _ completion:) | ``Purchases/purchase(package:completion:)`` |
+| restoreTransactions(_ completion:) | ``Purchases/restorePurchases(completion:)`` |
+| syncPurchases(_ completion:) | ``Purchases/syncPurchases(completion:)`` |
+| paymentDiscount(for:product:completion:) | Can be accessed from ``StoreProduct/introductoryPrice`` |
+| purchaseProduct(_:discount:_) | ``Purchases/purchase(product:discount:completion:)`` |
+| purchasePackage(_:discount:_) | ``Purchases/purchase(package:discount:completion:)`` |
```
Comment (aboedo) [inline Documentation.docc/V4_API_Migration_guide.md:154]: for Swift types, I added links to the actual types so you can click on them if you're viewing this as Docc (through Xcode's package documentation). If we deployed this as Docc or a separate Jazzy article, the link should work as well. 
The main advantage, though, is that we get type checks from Xcode here so we don't mess up a symbol name

File: Documentation.docc/V4_API_Migration_guide.md (hunk 1 above)
Comment (NachoSoto) [reply Documentation.docc/V4_API_Migration_guide.md:65]: It also so much easier to maintain. Thanks for taking the time to do this!

**Review by NachoSoto — APPROVED** [review]
😍

**Pull request merged**